formats/opentype.fathom

```diff
@@ -185,6 +186,197 @@ let language_id =
+        /// Last glyph ID in the range
+        end_glyph_id <- u16be,
+        /// Applied to all glyphs in the range
+        class <- u16be,
+    };
+
+    {
+        /// Number of ClassRangeRecords
+        class_range_count <- u16be,
+        /// Array of ClassRangeRecords — ordered by startGlyphID
+        class_range_records <- array16 class_range_count class_range_record,
+    }
+);
+
+/// # Class Definition Table
+///
+/// ## References
+///
+/// - [Microsoft's OpenType Spec: Class Definition Table](https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#class-definition-table)
+let class_def = {
+    /// Format identifier
+    class_format <- u16be,
+    /// Format specific data
+    data <- match class_format {
+        1 => class_def_format_1,
+        2 => class_def_format_2,
+        _ => fail,
+    },
+};
+
+/// # Coverage Format 1
+///
+/// ## References
+///
+/// - [Microsoft's OpenType Spec: Coverage Format 1](https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#coverage-format-1)
+let coverage_format_1 = {
+    /// Number of glyphs in the glyph array
+    glyph_count <- u16be,
+    /// Array of glyph IDs — in numerical order
+    glyph_array <- array16 glyph_count u16be,
+};
+
+/// # Coverage Format 2
+///
+/// ## References
+///
+/// - [Microsoft's OpenType Spec: Coverage Format 2](https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#coverage-format-2)
+let coverage_format_2 = (
+    /// RangeRecord
+    let range_record = {
+        /// First glyph ID in the range
+        start_glyph_id <- u16be,
+        /// Last glyph ID in the range
+        end_glyph_id <- u16be,
+        /// Coverage Index of first glyph ID in range
+        start_coverage_index <- u16be,
+    };
+
+    {
+        /// Number of RangeRecords
+        range_count <- u16be,
+        /// Array of glyph ranges — ordered by startGlyphID
+        range_records <- array16 range_count range_record,
+    }
+);
+
+/// # Coverage Table
+///
+/// ## References
+///
+/// - [Microsoft's OpenType Spec: Coverage Table](https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#coverageTbl)
+let coverage_table = {
+    /// Format identifier
+    coverage_format <- u16be,
+    /// Format specific data
+    data <- match coverage_format {
+        1 => coverage_format_1,
+        2 => coverage_format_2,
+        _ => fail,
+    }
+};
+
+/// # Device and VariationIndex Tables
+///
+/// Device tables and VariationIndex tables are used to provide adjustments to font-unit values in
+/// GPOS, JSTF, GDEF or BASE tables, such as the X and Y coordinates of an attachment anchor
+/// position.
+///
+/// Curiously the table has two interpretations. The second interprtation appears to be have been
+/// tacked on for variable fonts. The gist being that if the delta format is 0x8000 then the table
+/// is a VariationIndex table, which names the fields differently and does not contain a delta
+/// value array. E.g.
```

Comment:
Ahhhh, I get it now! It's not particularly elegant, but this does seems like a place where an `overlap` format could be used:

```
overlap {
    // Initial pass to figure out the table format
    init <- {
        _skipped <- array8 4 u8,
        table_format <- u16be,
    },
    // Device and VariationIndex Tables
    table <- match init.table_format {
        0x0001 => device_table,
        0x0002 => device_table,
        0x0003 => device_table,
        0x8000 => variation_index_table,
        0x7FFC => unknown_table, // not sure if this was meant to be “anything up to`0x7FFC`”?
        _ => fail,
    },
}
```

Reply:
>         0x7FFC => unknown_table, // not sure if this was meant to be “anything up to`0x7FFC`”?

From the comment in the spec I think they mean that each of the bits in `0x7FFC` (as opposed to that particular value) are reserved for future use (and should be set to zero):

> For future use — set to 0